libtuf/src/main/scala/com/advancedtelematic/libtuf/crypt/TufCrypto.scala

```diff
@@ -200,62 +213,81 @@ object TufCrypto {
 
 }
 
-protected [crypt] class EdCrypto extends TufCrypto[EdKeyType.type] {
+protected [crypt] object ECCrypto {
+  sealed trait CurveId {
+    def parameterSpec: ECParameterSpec
+  }
+  case object Curve25519 extends CurveId {
+    override def parameterSpec: ECParameterSpec = {
+      val ecP = ECNamedCurveTable.getParameterSpec("curve25519")
+      new ECParameterSpec(ecP.getCurve, ecP.getG, ecP.getN, ecP.getH, ecP.getSeed)
```

Comment:
What do you mean? it's used below, line 240

Reply:
Why does Curve25519.parameterSpec create a new ECParameterSpec, but Prime256v1 doesn't?

Reply:
Because of the way BC works with that curve, we need to instantiate a new ECParameterSpec, cannot just relly on the default settings, they don't work.

Reply:
Ok...